ethcore/src/engines/clique/mod.rs

```diff
@@ -0,0 +1,643 @@
+///    if not found, trigger an back-fill from last checkpoint.
+/// 4. Engine calls `state.apply(header)` and record the new state.
+
+/// About executive_author()
+/// Clique use author field for voting, the real author is hidden in the extra_data field. So
+/// When executing transactions (in `enact_verified()`, it will calls engine.executive_author() and use that.
+
+/// How sealing works:
+/// 1. implement `engine.set_signer()`. on startup, if miner account was setup on config/cli,
+///    `miner.set_author()` which will eventually be pass to here.
+/// 2. make `engine.seals_internally()` return Some(true).
+/// 3. on Clique::new setup IOService that impalement an timer that just calls `engine.step()`,
+///    which just calls `engine.client.update_sealing()` which triggers generating an new block.
+/// 4. `engine.generate_seal()` will be called by miner, which should return either Seal::None or Seal:Regular.
+///   a. return `Seal::None` if no signer or signer is not authorized.
+///   b. if period == 0 and block has transactions -> Seal::Regular, else Seal::None
+///   c. if we INTURN, wait for at least `period` since last block, otherwise wait for an random using algorithm as
+///      specified in the EIP.
+/// 5. Miner will create new block, in process it will call several engine method, which they need to do following:
+///   a. `engine_open_header_timestamp()` must set timestamp correctly.
+///   b. `engine.populate_from_parent()` must set difficulty to correct value. NOTE: this is used both in SYNCing and
+///       SEALing code path, for now we use an ugly hack to differentiate.
+/// 6. Implement `engine.on_seal_block()`, which is the new hook that allow modifying header after block is locked.
+/// 7. `engine.verify_local_seal()` will later be called, then normal syncing code path will also be called to import
+///    the new block.
+
+use std::cmp;
+use std::collections::VecDeque;
+use std::sync::{Arc, Weak};
+use std::time;
+use std::time::Duration;
+use std::time::SystemTime;
+use std::time::UNIX_EPOCH;
+
+use ethereum_types::{Address, H160, H256, U256};
+use hash::KECCAK_EMPTY_LIST_RLP;
+use itertools::Itertools;
+use lru_cache::LruCache;
+use parking_lot::RwLock;
+use rlp::encode;
+
+use block::*;
+use client::{BlockId, EngineClient};
+use engines::{Engine, Seal};
+use engines::clique::util::{extract_signers, recover_creator};
+use error::Error;
+use ethkey::Signature;
+use io::IoService;
+use machine::{Call, EthereumMachine};
+use types::BlockNumber;
+use types::header::{ExtendedHeader, Header};
+
+use super::signer::EngineSigner;
+
+use self::block_state::CliqueBlockState;
+use self::params::CliqueParams;
+use self::step_service::StepService;
+use std::collections::HashMap;
+use rand::Rng;
+
+mod params;
+mod block_state;
+mod step_service;
+mod util;
+
+// protocol constants
+/// Fixed number of extra-data prefix bytes reserved for signer vanity
+pub const SIGNER_VANITY_LENGTH: usize = 32;
```

Comment:
Do all of these consts need to be public? If that is indeed the case, they should all have documentation.

Reply:
done.they are all protocol consants, which is used in several files.